core/vm/logger.go

```diff
@@ -334,10 +334,11 @@ func (l *StructLogger) CaptureExit(output []byte, gasUsed uint64, err error) {
 
 	theLogPos := l.callStackLogInd[stackH-1]
 	l.callStackLogInd = l.callStackLogInd[:stackH-1]
-	theLog := l.logs[theLogPos]
+	theLog := &l.logs[theLogPos]
 	// update "forecast" data
 	if err != nil {
 		theLog.ExtraData.CallFailed = true
+		theLog.Err = err
```

Comment:
I think theLog should has a `*StructLogRes` type, which owns `Error` field rather than `Err`?